src/ucs/sys/topo/base/topo.c

```diff
@@ -207,34 +207,52 @@ ucs_topo_bus_id_to_sysfs_path(const ucs_sys_bus_id_t *bus_id, char *path,
                               size_t max)
 {
     const size_t prefix_length = strlen(UCS_TOPO_SYSFS_PCI_PREFIX);
-    char link_path[PATH_MAX];
+    ucs_status_t status        = UCS_OK;
+    char *link_path;
+
+    status = ucs_alloc_path_buffer(&link_path, "link_path");
+    if (status != UCS_OK) {
+        return status;
+    }
 
     if (max < PATH_MAX) {
-        return UCS_ERR_BUFFER_TOO_SMALL;
+        status = UCS_ERR_BUFFER_TOO_SMALL;
+        goto out_free_buffer;
     }
 
-    ucs_strncpy_safe(link_path, UCS_TOPO_SYSFS_PCI_PREFIX, sizeof(link_path));
+    ucs_strncpy_safe(link_path, UCS_TOPO_SYSFS_PCI_PREFIX, PATH_MAX);
     ucs_topo_bus_id_str(bus_id, 0, link_path + prefix_length,
                         PATH_MAX - prefix_length);
     if (realpath(link_path, path) == NULL) {
-        return UCS_ERR_IO_ERROR;
+        status = UCS_ERR_IO_ERROR;
     }
 
-    return UCS_OK;
+out_free_buffer:
+    ucs_free(link_path);
+    return status;
 }
 
 static int
 ucs_topo_read_device_numa_node(const ucs_sys_bus_id_t *bus_id)
 {
-    char path[PATH_MAX];
+    int numa_node = 0;
     ucs_status_t status;
+    char *path;
+
+    status = ucs_alloc_path_buffer(&path, "sysfs_path");
+    if (status != UCS_OK) {
+        return status;
+    }
 
-    status = ucs_topo_bus_id_to_sysfs_path(bus_id, path, sizeof(path));
+    status = ucs_topo_bus_id_to_sysfs_path(bus_id, path, PATH_MAX);
     if (status != UCS_OK) {
+        ucs_free(path);
```

Comment:
Better `goto` -> label, to ensure correct order..

Reply:
Unresolved

Reply:
done